lisa/wa.py

```diff
@@ -552,3 +553,61 @@ def __init__(self, wa_output, trace_to_df, **kwargs):
     def _get_artifact_df(self, path):
         trace = Trace(path, **self._trace_kwargs)
         return self._trace_to_df(trace)
+
+class WAJankbenchCollector(WAArtifactCollectorBase):
+    """
+    WA collector for the jankbench frame timings.
+
+    The collector framework will return a single :class:`pandas.DataFrame` with the
+    results from every jankbench job in WACollector database format (i.e. the returned dataframe
+    is arranged such that each reported metric is separated as a separate row). The metrics
+    reported are:
+
+    . ``total_duration``: Time in milliseconds to complete the frame
+
+    . ``jank_frame``: Boolean indicator of missed frame deadline. ``1`` is a Jank frame, ``0`` is not.
+
+    . ``name``: Subtest name, provided by the Jankbench app
+
+    . ``frame_id``: monotonically increasing frame number, starts from ``1`` for each subtest iteration.
+
+    An example plotter matching the old-style output can be found in the jupyter notebook
+    working directory at :file:`ipynb/wltests/WAOutput-JankbenchDemo.ipynb`
+
+    If you have existing code expecting a more direct translation of the original sqlite database
+    format, you can massage the collected dataframe back into a closer resemblance to the
+    original source database with this sequence of pandas operations::
+
+        collector = wa.get_collector('jankbench')
+        df = collector.df
+        db_df = df.pivot(index=['iteration', 'id', 'kernel', 'frame_id'], columns=['variable'])
+        db_df = db_df['value'].reset_index()
+        db_df.columns.name = None
+        # db_df now looks more like the original format
+
+    """
+
+    NAME = 'jankbench'
+    _EXPECTED_WORKLOAD_NAME = 'jankbench'
+    _ARTIFACT_NAME = "jankbench-results"
+    _ARTIFACT_METRIC_SUFFIX_UNIT = defaultdict(
+        # Default is an empty string
+        str,
+        {
+            'total_duration': 'ms',
+            'jank_frame': 'boolean',
+        },
+    )
+
+    def _get_artifact_df(self, path):
+        con = sqlite3.connect(path)
```

Comment:
would probably be a good idea to close the connection once we are done. The best way to do it would be to use:
https://docs.python.org/3/library/contextlib.html#contextlib.closing

```
from contextlib import closing
with closing(sqlite3.connect(path)) as con:
    raw_df = pd.read_sql_query("SELECT total_duration, jank_frame, name, _id as frame_id from ui_results", con) 

df = raw_df.melt(id_vars=['frame_id'], value_vars=['total_duration', 'jank_frame'])
return df 
```